### PR TITLE
Test fix: ensure we don't accidentally generate two identical histograms

### LINKED ISSUE #113264: [CI] CCSTelemetrySnapshotTests.testEqualsAndHashcode
### CI Link

https://gradle-enterprise.elastic.co/s/wfr3ymfwqe44k/tests/task/:server:test/details/org.elasticsearch.action.admin.cluster.stats.CCSTelemetrySnapshotTests/testEqualsAndHashcode?top-execution=1

### Repro line

./gradlew ":server:test" --tests "org.elasticsearch.action.admin.cluster.stats.CCSTelemetrySnapshotTests.testEqualsAndHashcode" -Dtests.seed=20E3FD82C7E52278 -Dtests.locale=ps -Dtests.timezone=Japan -Druntime.java=22	

### Does it reproduce?

Yes

### Applicable branches

main, 8.x

### Failure history

_No response_

### Failure excerpt

Happened on a PR, but also reproduces on main.

### PR DESCRIPTION
**Pull request by smalyshev**
Fixes https://github.com/elastic/elasticsearch/issues/113264

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/action/admin/cluster/stats/CCSTelemetrySnapshotTests.java [hunk 1]
```diff
@@ -33,7 +33,7 @@ public class CCSTelemetrySnapshotTests extends AbstractWireSerializingTestCase<C
 
     private LongMetricValue randomLongMetricValue() {
         LongMetric v = new LongMetric();
-        for (int i = 0; i < randomIntBetween(1, 10); i++) {
+        for (int i = 0; i < randomIntBetween(5, 10); i++) {
```
Comment (andreidan) [inline server/src/test/java/org/elasticsearch/action/admin/cluster/stats/CCSTelemetrySnapshotTests.java:36]: It's not immediately obvious what issue we're looking to fix and how this change fixes it. 

Can you please expand in the PR description to observed problem? 

I suspect the problematic usage of this random long method lies in `mutateInstance` ? (based on the link issue) If this is correct I think we should use `randomValueOtherThan()` to make sure `mutateInstance` generates a value that's different than the input.

**Review by andreidan — COMMENTED** [review]
Thanks for working on this Stas.

Left a suggestion to potentially improve the reliability of the solution.

File: server/src/test/java/org/elasticsearch/action/admin/cluster/stats/CCSTelemetrySnapshotTests.java (hunk 1 above)
Comment (smalyshev) [reply server/src/test/java/org/elasticsearch/action/admin/cluster/stats/CCSTelemetrySnapshotTests.java:36]: @andreidan I changed the code to use `randomValueOtherThan`

**Review by andreidan — APPROVED** [review]
LGTM, thanks for fixing this Stas

**Comment by smalyshev** [comment]
@elasticmachine update branch

**Pull request merged**